### PR TITLE
chore: increase default cdk version and replace deprecated methods

### PR DESCRIPTION
**Pull request by henrysachs**
I increased the cdk version to match the yesterday released one and also replaced the deprecated function call from `addPeerDependencies`. Just some cleanups that help me in my custom Projen module.

### CODE REVIEW

**Comment by henrysachs** [comment]
if its fine for all i maybe take a look at other deprecated functions and try to convert them.

**Comment by henrysachs** [comment]
i now went through all files and replaced deprecated methods without breaking compatibility. That's why i added string to the Type aswell as Semver.

**Comment by henrysachs** [comment]
The License File Test seems to be a bit flaky isn't it? it seems to sometimes fail and sometimes pass. 🤔

File: src/cdk8s-construct.ts [hunk 1]
```diff
@@ -25,10 +24,10 @@ export class ConstructLibraryCdk8s extends ConstructLibrary {
 
     const ver = options.cdk8sVersion;
 
-    this.addPeerDependencies({
-      'constructs': Semver.caret('2.0.2'),
-      'cdk8s': Semver.caret(ver),
-      'cdk8s-plus': Semver.caret(ver),
-    });
+    this.addPeerDeps(
+      'constructs@^2.0.2',
```
Comment (eladb) [inline src/cdk8s-construct.ts:28]: Let's also bump cdk8s to the latest version and I believe it now uses the same constructs major version as the CDK.

File: src/jsii-docgen.ts [hunk 2]
```diff
@@ -10,7 +9,7 @@ import { StartEntryCategory } from './start';
  */
 export class JsiiDocgen {
   constructor(project: JsiiProject) {
-    project.addDevDependencies({ 'jsii-docgen': Semver.caret('1.3.2') });
+    project.addDevDeps('jsii-docgen@^1.3.2');
```
Comment (eladb) [inline src/jsii-docgen.ts:12]: I don't think a version is needed here. Just omit the version

File: src/jsii-project.ts [hunk 3]
```diff
@@ -249,15 +252,15 @@ export class JsiiProject extends TypeScriptProject {
...
 
-    this.addDevDependencies({
-      'jsii': jsiiVersion,
-      'jsii-diff': jsiiVersion,
-      'jsii-pacmak': jsiiVersion,
-      'jsii-release': Semver.caret('0.1.6'),
-      '@types/node': Semver.caret(minNodeVersion),
-    });
+    this.addDevDeps(
+      `jsii@^${jsiiVersion}`,
```
Comment (eladb) [inline src/jsii-project.ts:258]: I think we can completely omit jsii version.

File: src/typescript-typedoc.ts [hunk 4]
```diff
@@ -7,7 +6,7 @@ import { TypeScriptProject } from './typescript';
  */
 export class TypedocDocgen {
   constructor(project: TypeScriptProject) {
-    project.addDevDependencies({ typedoc: Semver.caret('0.17.8') });
+    project.addDevDeps('typedoc@^0.17.8');
```
Comment (eladb) [inline src/typescript-typedoc.ts:9]: Omit version

File: src/typescript.ts [hunk 5]
```diff
@@ -26,7 +26,7 @@ export interface TypeScriptProjectOptions extends NodeProjectOptions {
    * TypeScript version to use.
    * @default "^3.9.5"
    */
-  readonly typescriptVersion?: Semver;
+  readonly typescriptVersion?: string | Semver;
```
Comment (eladb) [inline src/typescript.ts:29]: Unions are Not recommended in jsii

**Review by eladb — CHANGES_REQUESTED** [review]
Generally with the  new "deps" options you don't really need a version specifier

**Comment by henrysachs** [comment]
will now pick up the requested changed

File: src/typescript.ts (hunk 5 above)
Comment (henrysachs) [reply src/typescript.ts]: i now moved everything to string but wouldn't that create a breaking change for everyone using this property?

File: src/jsii-project.ts (hunk 3 above)
Comment (henrysachs) [reply src/jsii-project.ts]: I'll omitted the version but kept the parameter, if a user want to set it to their own fixed version.

**Comment by henrysachs** [comment]
i now removed jsii version as a parameter completly and therefore omitted the version

File: src/jsii-project.ts [hunk 6]
```diff
@@ -249,15 +245,13 @@ export class JsiiProject extends TypeScriptProject {
...
-      'jsii-pacmak': jsiiVersion,
-      'jsii-release': Semver.caret('0.1.6'),
-      '@types/node': Semver.caret(minNodeVersion),
-    });
+    this.addDevDeps(
+      'jsii',
+      'jsii-diff',
+      'jsii-pacmak',
+      'jsii-release',
+      `@types/node@^${minNodeVersion}`,
```
Comment (eladb) [inline src/jsii-project.ts:253]: I suspect this is already covered by `TypeScriptProject` and therefore not required here anymore

File: src/node-project.ts [hunk 7]
```diff
@@ -787,8 +787,8 @@ export class NodeProject extends Project {
 
     const projen = options.projenDevDependency ?? true;
     if (projen) {
-      const projenVersion = options.projenVersion ?? Semver.caret(PROJEN_VERSION);
-      this.addDevDependencies({ projen: projenVersion });
+      const projenVersion = options.projenVersion ?? `^${PROJEN_VERSION}`;
+      this.addDevDeps( `projen@${projenVersion}` );
```
Comment (eladb) [inline src/node-project.ts:791]:
```suggestion
      this.addDevDeps(`projen@${projenVersion}`);
```

**Comment by henrysachs** [comment]
removed the min node version from jsii

**Review by eladb — APPROVED** [review]

**Pull request merged**